### ISSUE TITLE
Issue trying to connect Remix with Besu making use of RPC-HTTP connection

### ISSUE THREAD

**Issue by a905284** [hyperledger/besu#2187]
### Description

I am trying to connect a Besu node with a Remix smart contract, making use of the Web3 Provider of Remix. I am following an [example ](https://besu.hyperledger.org/en/stable/HowTo/Get-Started/Starting-node/) provided in the Hyperledger Besu official website, but I am not able to make the connection. 

I am making use of an AWS EC2 ubuntu instance and Docker. The docker compose file can be found attached, together with an screenshot proving that I am actually allowing AWS to listen on port 8545 and 8546, which correspond to RPC-HTTP and WebSocket connections.

When trying to connect to the port 8545 (I attach an screenshot of how I am doing so on Remix), I get an error message which I also attach below. I am checking the logs of the Besu container, but I don't get to see any actual logs corresponding to this connection. I am also attaching a screenshot of the starting logs of Besu, in which it can be seen that Besu is actually listen for RPC-HTTP connections on the port 8545.

### Steps to Reproduce (Bug)
1. I am first running the docker-compose.yaml file on the VM machine in VSCode, making use of a Remote SSH connection to AWS.
2. Afterwards I go to remix, and I try to connect with the Besu node

**Expected behavior:** [What you expect to happen]

The expected behavior is that Remix would connect to the Besu node on port 8545.

**Actual behavior:** [What actually happens]

The actual behavior is that Remix is not able to connect to the Besu node, and I get no logs on the Besu docker container.

**Frequency:** [What percentage of the time does it occur?]

It happens all the time. I haven't been able to make the connection even once.

### Versions (Add all that apply)
* Software version:  Version 3
* OS Name & Version:  Windows 10
* Virtual Machine software & version: Ubuntu 18.04
* Docker Version: latest
* Cloud VM, type, size: AWS EC2 t2.medium instance.

![Ports AWS](https://user-images.githubusercontent.com/78016113/115996485-bc472000-a5df-11eb-84b5-95969ce1cfb2.PNG)
![Remix](https://user-images.githubusercontent.com/78016113/115996488-bc472000-a5df-11eb-9233-202707585d5a.PNG)
![Remix_Error](https://user-images.githubusercontent.com/78016113/115996490-bcdfb680-a5df-11eb-9cc4-abd9e6e27e2e.PNG)
![Besu container](https://user-images.githubusercontent.com/78016113/115996491-bd784d00-a5df-11eb-85b8-95a31a1e2951.PNG)
![BesuStartingLogs](https://user-images.githubusercontent.com/78016113/115996492-bd784d00-a5df-11eb-9350-b3c338e1a623.PNG)

**Comment by macfarla** [comment]
I see you have added http://remix.ethereum.org to the cors but the recommendation is to add https://remix.ethereum.org - have you tried that?

**Comment by sajz** [comment]
@a905284 - did @macfarla rec fix your issue?

Closing for now, reopen if it's still an issue.

**Comment by garyschulte** [comment]
FWIW, I recently was using besu with remix and figured I would add some color to this issue.  I *was* able to connect to a local besu instance using the "Web3 provider" URL, but when it connects, remix does not have any accounts to use.  For this option remix is expecting a geth node which has a local account.  Besu does not have a concept of local accounts and it expects all transactions sent through it to be pre-signed.

The best way I found to use besu with remix is to use the "Injected Web3 Provider" option, with metamask.  I configured metamask to point to my local besu instance and it worked seamlessly (and allows you to use any private key you have configured with metamask)
